core/primitives/src/block.rs

```diff
@@ -522,38 +521,59 @@ impl Block {
     }
 }
 
-#[derive(Clone)]
-pub enum MaybeNew<'a, T> {
-    New(&'a T),
-    Old(&'a T),
+#[derive(Clone, Debug)]
+pub enum ChunkType<'a> {
+    New(&'a ShardChunkHeader),
+    Old(&'a ShardChunkHeader),
```

Comment:
Can you add a comment about the issue where old chunk's data (shard id in particular) may not be compatible with current protocol version? I'm not sure what's the best place to put it, up to you.